### ISSUE TITLE
css fix for user experience

### ISSUE THREAD

**Issue by satra** [dandi/dandiarchive#636]
would it be possible to fix the css/vuetify controls such that the top row and each individual item in the list fit the view.

![image](https://user-images.githubusercontent.com/184063/115595966-90f1c780-a2a5-11eb-92a0-7d0c2c84d05e.png)

**Comment by waxlamp** [comment]
Is this a specific concern for mobile? And is the problem that the text runs off the right edge, rather than wrapping?

**Comment by satra** [comment]
generally for any narrow viewframe, not just mobile. and yes, the issue is that some specific containers are not responsive.

**Comment by waxlamp** [comment]
@jtomeck could you lay some CSS on us?

**Comment by jtomeck** [comment]
I'm not sure this is a CSS issue, as Vuetify has it's own responsiveness classes baked in. That said, I believe there are more issues that need discussion than the content bleeding off the page. What I mean is, if we were to utilize Vuetify's responsive grid, so that the content "falls in line" on mobile devices, we need answers to the following questions:

1. Is all information necessary at a glance?
    - Can some be hidden until tapped (ie. a button to expand the row to show the metadata?)
    - If not, how will the metadata re-organize itself so that the list is still usable?
2. Will titles break or truncate with ellipsis?
3. Can padding/spacing be reduced?

Once we have answers/solutions to these things I think I could potentially hop in there can get this all fixed up. NOt sure if @AlmightyYakob has thoughts on working with responsiveness with vuetify's grid system either, but let me know!

**Comment by waxlamp** [comment]
@satra can likely answer those questions more accurately than I can, but for the sake of argument, supposing the answers are:

1. Yes
  - No
  - N/A
 2. Break
 3. Yes, but not at the cost of legibility/usability

then what would your general approach be here?

**Comment by jtomeck** [comment]
**Page structure:**
1. Remove a lot of left/right padding

**List items:**
1. Shrink "draft" badge
2. Wrap titles
3. Abbreviate month
4. What does "contact" mean? Can it be removed?

---

Along with everything mentioned above, it would be using Vuetify's grid with responsive classes to make sure everything collapses well. I don't really know how to explain the approach better than that without just going in and doing it.

**Comment by waxlamp** [comment]
@jtomeck, what's the expected level of effort if you got into this?

@satra, can you confirm/modify the prospective answers I gave to Jared's questions [above](https://github.com/dandi/dandiarchive/issues/636#issuecomment-846117791)?

**Comment by satra** [comment]
> Is all information necessary at a glance?
> Can some be hidden until tapped (ie. a button to expand the row to show the metadata?)

Since the intent of the responsiveness is about narrower views, the most important things are likely: Title, version, and size. contact info  (which doesn't show up anyway) and last updated can be removed. i suspect the latter will become a separate ui element about sorting the order.

> Will titles break or truncate with ellipsis?

ideally the title wraps around.

> Can padding/spacing be reduced?

yes.

**Comment by jtomeck** [comment]
> @jtomeck, what's the expected level of effort if you got into this?

@waxlamp - I'll take a little bit of time to go in and assess the situation. If we haven't been planning to code for all screen sizes from the beginning, it could be a bigger haul than anticipated.

**Comment by jtomeck** [comment]
At a glance, it doesn't seem like it'll be a large haul. It will be throwing some CSS at the list items primarily as well as some slight size changes/reworks here and there. Maybe 2 hours for this?

I also noticed on the dandiset landing page, the meta sidebar is above the main content body when the window is sized down. I'd like to figure something out for that as well. This part might be a little work, because I am thinking it would probably be good to still show this information at the top. It just needs to be made more compact for smaller screens.

**Comment by waxlamp** [comment]
> At a glance, it doesn't seem like it'll be a large haul. It will be throwing some CSS at the list items primarily as well as some slight size changes/reworks here and there. Maybe 2 hours for this?

Great! Please go ahead and do the work.

> 
> I also noticed on the dandiset landing page, the meta sidebar is above the main content body when the window is sized down. I'd like to figure something out for that as well. This part might be a little work, because I am thinking it would probably be good to still show this information at the top. It just needs to be made more compact for smaller screens.

I fixed this in another PR just today. The fix I came up with was very simple: just move the panel content to below the main content. This was the easiest way to solve the problem of metadata-on-top-on-small-screens, and it's ok for now. FWIW, I requested your review on that PR; if you check your notifications, you can still leave a review (even though it's already merged--if you have a better solution, we can still use it).

**Comment by jtomeck** [comment]
> FWIW, I requested your review on that PR

Ah okay sorry about that. I'm glad you didn't wait on my review to merge so I wasn't holding anything up.

**Comment by waxlamp** [comment]
> > FWIW, I requested your review on that PR
> 
> Ah okay sorry about that. I'm glad you didn't wait on my review to merge so I wasn't holding anything up.

No problem. I was more pointing you there so you can see (and possibly comment on) the fix I put in.
